packages/lisk-transaction-pool/src/job.ts

```diff
@@ -0,0 +1,50 @@
+export class Job<T> {
+	private _id: NodeJS.Timer | undefined;
+	private readonly _interval: number;
+	private readonly _job: () => Promise<T>;
+	private _running = false;
+
+	public constructor(
+		job: () => Promise<T>,
+		interval: number,
+	) {
+		this._interval = interval;
+		this._job = job;
+	}
+
+	public async start(): Promise<void> {
+		if (!this._running) {
+			this._running = true;
+
+			return this.run();
+		}
+	}
+
+	public stop(): void {
+		if (this._running && this._id  !== undefined) {
+			clearTimeout(this._id);
+			this._id = undefined;
+			this._running = false;
+		}
+	}
+
+	private async callJobAfterTimeout(): Promise<void> {
+		return new Promise<void>(resolve => {
+			this._id = setTimeout(async () => {
+				await this._job();
+
+				return resolve();
+			}, this._interval);
+		});
+	}
+
+	private async run(): Promise<void> {
+		// Base case for recursive function
+		if (!this._running) {
+			return;
+		}
+		await this.callJobAfterTimeout();
+
+		return this.run();
+	}
+}
```

Comment:
Don't you have an agreement of empty line at the end of files?

Reply:
I guess a standard, as you see GitHub is marking it as an error by default. Lisk Core has it in every file.

Reply:
`npm run format` should solve the problem.
Prettier is doing it automatically